File: GestionEvenementBack/src/main/java/tn/eesprit/gestionevenementback/Entities/Goods.java
package tn.eesprit.gestionevenementback.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import javax.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class Goods implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer GoodsId;
    String NameGoods;
    Integer Quantity;
    Float UnitPrice;
    Integer Stock;



    @JsonIgnore
    @ManyToMany
    Set<ResourceMateriel> resources;

}

File: GestionEvenementFront/src/app/_Models/Goods.ts
import {Resource} from "@angular/compiler-cli/src/ngtsc/metadata";
import {ResourceMateriel} from "./ResourceMateriel";

export class Goods {
  goodsId: string;
  NameGoods:string;
  Quantity:string ;
  UnitPrice:string ;
  resources:ResourceMateriel[];
  stock:string;
}
